### PR TITLE
[ci] Use current PHP_BINARY when running ./phpunit

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

This allows to easilly test with several php versions.
I also added an auto-install of composer and the deps when they are not here.
Should ease on-boarding devs to running tests.

### CODE REVIEW

File: phpunit [hunk 1]
```diff
@@ -6,12 +6,26 @@ use Symfony\Component\Process\ProcessUtils;
...
+            fopen('https://getcomposer.org/composer.phar', 'rb'),
+            fopen($COMPOSER = __DIR__.'/composer.phar', 'wb')
+        );
+    }
+}
 
-// PHPUnit 4.8 does not support PHP 7, while 5.0 requires PHP 5.6+
-if (PHP_VERSION_ID >= 70000) {
-    $PHPUNIT_VERSION = '5.0';
+$PHP = ProcessUtils::escapeArgument($PHP);
```
Comment (stof) [inline phpunit]: will this work properly on HHVM ? Or are they exposing it as ``hhvm --php`` ?

File: phpunit [hunk 2]
```diff
@@ -6,12 +6,26 @@ use Symfony\Component\Process\ProcessUtils;
...
+$COMPOSER = $PHP.' '.ProcessUtils::escapeArgument($COMPOSER);
+
+if (!file_exists(__DIR__.'/vendor')) {
+    passthru("$COMPOSER install --no-progress --ansi");
```
Comment (stof) [inline phpunit]: shouldn't it use ``update``, to be sure that the dependencies are uptodate ?

File: phpunit (hunk 1 above)
Comment (nicolas-grekas) [inline phpunit]: PHP_BINARY will not contain `--php` so it's ok

File: phpunit (hunk 2 above)
Comment (nicolas-grekas) [inline phpunit]: changed

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**